File: problems/leetcode/leetcode-1423-maximum-points-you-can-obtain-from-cards/main.cc
#include <numeric>
#include <vector>
using namespace std;
class Solution {
   public:
    int maxScore(vector<int>& a, int k) {
        int n = a.size(), k1 = n - k;
        int s = accumulate(a.begin(), a.end(), 0);
        int min_sum = s, sum = 0;

        for (int L = 0, R = 0; R < n; R++) {
            sum += a[R];
            while (R - L + 1 > k1) sum -= a[L++];
            if (R - L + 1 == k1) min_sum = min(min_sum, sum);
        }
        return  s- min_sum;
    }
};

File: problems/leetcode/leetcode-1423-maximum-points-you-can-obtain-from-cards/main.py
from typing import List


class Solution:
    def maxScore(self, a: List[int], k: int) -> int:
        k1 = len(a) - k
        S = sum(a)
        min_s = S
        s = 0

        L = R = 0
        while R < len(a):
            s += a[R]
            while R - L + 1 > k1:
                s -= a[L]
                L += 1
            if R - L + 1 == k1:
                if min_s > s:
                    min_s = s
            R+=1
        return S - min_s
